cupy/random/_generator.py

```diff
@@ -809,8 +809,7 @@ def seed(self, seed=None):
             if isinstance(seed, numpy.ndarray):
                 seed = int(hashlib.md5(seed).hexdigest()[:16], 16)
             else:
-                seed = int(
-                    numpy.asarray(seed).astype(numpy.uint64, casting='safe'))
+                seed = int(numpy.asarray(seed, dtype=numpy.uint64))
```

Comment:
This changes the behavior; the new code implicitly casts `cupy.random.seed(1.23)` to uint64, but it should raise an error, like numpy does.

Reply:
The NumPy code actually checks ranges manually, an alternative would be to do:
```
seed_passed = numpy.asarray(seed)
# Cast allow integer overflows:
seed = seed_passed.astype(numpy.uint64, casting='same_kind')
if (seed != seed_passed).any():
    raise ValueError("Seed must be between 0 and 2**64 - 1")
```
(NumPy actually requires 32 bit seeds, but maybe no point in changing behavior here).

Reply:
Really, sorry... the `same_kind` casting doesn't work, maybe this:
```
                seed_arr = numpy.asarray(seed)
                if seed_arr.dtype.kind not in "iu":
                    raise TypeError("Seed must be an integer.")
                seed = int(seed_arr)
                # Check that no integer overflow occurred during the cast
                if seed < 0 or seed >=2**64:
                    raise ValueError('Seed must be an integer between 0 and 2**64 - 1')
```

I would suggest just replacing the whole thing with `seed = operator.index()` and the range check, but I am not 100% if that might create some other regression.